spec/specification/events_spec.rb

```diff
@@ -0,0 +1,222 @@
+
+      # All handlers should have been called despite the failure
+      expect(handler1_called).to be true
+      expect(handler2_called).to be true
+      expect(handler3_called).to be true
+
+      # Cleanup
+      OpenFeature::SDK.remove_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler1)
+      OpenFeature::SDK.remove_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, failing_handler)
+      OpenFeature::SDK.remove_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler2)
+      OpenFeature::SDK.remove_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler3)
+    end
+  end
+
+  context "Requirement 5.2.6" do
+    specify "Event handlers MUST persist across provider changes" do
+      # Wait for initial provider to be ready
+      sleep(0.1)
+
+      handler_call_count = 0
+      handler = ->(_event_details) { handler_call_count += 1 }
+
+      # Add handler after initial provider is already set and ready
+      OpenFeature::SDK.add_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler)
+
+      # Set first provider
+      provider1 = OpenFeature::SDK::Provider::InMemoryProvider.new
+      OpenFeature::SDK.set_provider(provider1)
+      sleep(0.1)
+
+      expect(handler_call_count).to eq(1)
+
+      # Set second provider - handler should still be active
+      provider2 = OpenFeature::SDK::Provider::NoOpProvider.new
+      OpenFeature::SDK.set_provider(provider2)
+      sleep(0.1)
+
+      expect(handler_call_count).to eq(2)
+
+      # Cleanup
+      OpenFeature::SDK.remove_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler)
+    end
+  end
+
+  context "Requirement 5.3.1" do
+    specify "If the provider's initialize function terminates normally, PROVIDER_READY handlers MUST run" do
+      ready_event_received = false
+      handler = ->(_event_details) { ready_event_received = true }
+
+      OpenFeature::SDK.add_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler)
+
+      # Provider with successful init
+      provider = OpenFeature::SDK::Provider::InMemoryProvider.new
+      allow(provider).to receive(:init).and_return(nil) # Normal termination
+
+      OpenFeature::SDK.set_provider(provider)
+      sleep(0.1) # Wait for async initialization
+
+      expect(ready_event_received).to be true
+
+      # Cleanup
+      OpenFeature::SDK.remove_handler(OpenFeature::SDK::ProviderEvent::PROVIDER_READY, handler)
+    end
+  end
+
+  context "Requirement 5.3.3" do
+    specify "Handlers attached after the provider is already in the associated state, MUST run immediately" do
+      # NOTE: This requirement is about handlers running immediately when attached after a provider
+      # is already in the associated state (e.g., READY). This feature is not yet implemented
+      # in the Ruby SDK. Handlers are only triggered by state transitions, not by current state.
+      skip "Immediate handler execution for current state not yet implemented"
+    end
+  end
```

Comment:
I think we need to implement this ass well before a release. If this PR is getting too big, I don't mind if it gets done in a separate one, but I don't want it to linger un-implemented too long.